cmd/geth/config.go

```diff
@@ -189,7 +191,20 @@ func makeFullNode(ctx *cli.Context) (*node.Node, ethapi.Backend) {
 	if ctx.IsSet(utils.SyncTargetFlag.Name) && cfg.Eth.SyncMode == downloader.FullSync {
 		utils.RegisterFullSyncTester(stack, eth, ctx.Path(utils.SyncTargetFlag.Name))
 	}
-	return stack, backend
+
+	// Start the dev mode if requested
```

Comment:
You can also define a function `RegisterBeaconSim` or so in `utils` package and `Fatalf` if any error is occurred.
So that we don't need to return the error.

Reply:
I modified it to not return an error (it seems that's what the code here already did before, so I guess it makes sense not to change that).  I think it's not really that much code so I didn't put it into its own function.